pandas/_libs/tslibs/util.pxd

```diff
@@ -227,3 +236,37 @@ cdef inline bint is_nan(object val):
     is_nan : bool
     """
     return (is_float_object(val) or is_complex_object(val)) and val != val
+
+
+cdef inline const char* get_c_string_buf_and_size(object s,
```

Comment:
can the type for `s` be made more specific?

Only for if you need to make another pass through this file for other reasons: pls try to avoid 1-letter variable names.

Curious about your opinion: instead of `char*` we could use `p_char` (i.e. `from cython cimport p_char`).  Downside is it is less like native C, upside is it is more like native python.  Ditto for lots of places where we use `foo*`.

Reply:
Changed: s -> py_string

Reply:
IMHO that the pointer is used at a rather low level, and usually at this level work those who are familiar with the C, and for them it is better to have an explicit declaration than any of alias.

Reply:
As concerns more specific type for, already, `py_string`:
I tnink there's not reason because we have some checks in this function, that works for all python types. Thanks to them, in this particular situation we can use api, which is more efficient than the code generated by cython.

Reply:
> Downside is it is less like native C, upside is it is more like native python.

I think that in this particular case it would be better to see that this is actually a raw C-level buffer, so users should stay alert when working with it.
Also for added safety we made return type of `const char*`, and IMHO `const char_p` doesn't look any nicer or Pythonic :)